src/Filesystem/Provider/FileAccess.php

```diff
@@ -0,0 +1,257 @@
+<?php
+
+namespace ILIAS\Filesystem\Provider;
+
+use ILIAS\Filesystem\DTO\FileSize;
+use ILIAS\Filesystem\Exception\FileAlreadyExistsException;
+use ILIAS\Filesystem\Exception\FileNotFoundException;
+use ILIAS\Filesystem\Exception\IOException;
+
+/**
+ * Interface FileAccess
+ *
+ * The FileAccess interface defines all file operations.
+ * Filesystem role interface.
+ *
+ * @author  Nicolas Schäfli <ns@studer-raimann.ch>
+ * @since 5.3
+ * @version 1.0
+ *
+ * @public
+ */
+interface FileAccess {
+
+	/**
+	 * Writes the content to a new file.
+	 *
+	 * @param string $path      The path to the file which should be created.
+	 * @param string $content   The content which should be written to the new file.
+	 *
+	 * @return void
+	 *
+	 * @throws FileAlreadyExistsException   If the file already exists.
+	 * @throws IOException                  If the file could not be created or written.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function write($path, $content);
+
+
+	/**
+	 * Updates the content of a file.
+	 * Replaces the file content with a new one.
+	 *
+	 * @param string $path          The path to the file which should be updated.
+	 * @param string $newContent    The new file content.
+	 *
+	 * @return void
+	 *
+	 * @throws FileNotFoundException    If the file is not found.
+	 * @throws IOException              If the file could not be updated.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function update($path, $newContent);
+
+
+	/**
+	 * Creates a file or updates an existing one.
+	 *
+	 * @param string $path      The path to the file which should be created or updated.
+	 * @param string $content   The content which should be written to the file.
+	 *
+	 * @return void
+	 *
+	 * @throws IOException  If the file could not be created or updated.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function put($path, $content);
+
+
+	/**
+	 * Reads a file content to a string.
+	 *
+	 * @param string $path                  The path to the file which should be read.
+	 *
+	 * @return string   The file content.
+	 *
+	 * @throws FileNotFoundException        If the file doesn't exist.
+	 * @throws IOException                  If the file could not be red.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function read($path);
+
+
+	/**
+	 * Checks whether a file exists.
+	 *
+	 * @param string $path The file path which should be checked.
+	 *
+	 * @return bool True if the file exists, otherwise false.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function has($path);
```

Comment:
This comes from flysystem, the underlying library:

`This only has consistent behaviour for files, not directories. Directories are less important in Flysystem, they’re created implicitly and often ignored because not every adapter (filesystem type) supports directories.`

But I see the point. Checking for directories is used a lot in ILIAS. We should provide this here again.